packages/npm-dependencies-validation/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import { spawnCommand } from "./utils/npmUtil";
+import { resolve, dirname } from "path";
+import { isManagedByNpm6 } from "./utils/packageJsonUtil";
+import {
+  DependencyIssue,
+  IssueType,
+  NpmLsDependency,
+  VscodeFsUri,
+} from "./types";
+
+const LS_DEPS = ["ls", "--depth=0"]; // dependencies and extraneous modules
+const LS_DEV_DEPS = [...LS_DEPS, "--dev"]; // devDependencies
+
+async function getDepsStatus(
+  packageJsonPath: string,
+  devDeps: boolean
+): Promise<any> {
+  // try to use DependencyType ???
+  return spawnCommand(
```

Comment:
We will be executing more npm commands in the near future (`prune / install`).
And will have additional shared concerns while executing them:
- output redirection
- timeout
- ?

Perhaps there should be an `npmCommandsInvoker.ts` file to handle this concern.
This could also make it easy to refactor the implementation in the future, e.g if we switch to the npm's programatic APi.

Reply:
nevermind i thought the spawn is here directly.